src/main.ts

```diff
@@ -510,29 +509,33 @@ function makeLongUtils(options: Options, bytes: ReturnType<typeof makeByteUtils>
   const longToString = conditionalOutput(
     "longToString",
     code`
-      function longToString(long: ${Long}) {
-        return long.toString();
+      function longToString(int64: { toString(): string }) {
+        return int64.toString();
       }
     `,
   );
 
   const longToBigint = conditionalOutput(
     "longToBigint",
     code`
-      function longToBigint(long: ${Long}) {
-        return BigInt(long.toString());
+      function longToBigint(int64: { toString(): string }) {
+        return typeof int64 == "bigint" ? int64 : ${bytes.globalThis}.BigInt(int64.toString());
       }
     `,
   );
 
   const longToNumber = conditionalOutput(
     "longToNumber",
     code`
-      function longToNumber(long: ${Long}): number {
-        if (long.gt(${bytes.globalThis}.Number.MAX_SAFE_INTEGER)) {
+      function longToNumber(int64: { toString(): string }): number {
+        const num = ${bytes.globalThis}.Number(int64.toString());
+        if (num > ${bytes.globalThis}.Number.MAX_SAFE_INTEGER) {
           throw new ${bytes.globalThis}.Error("Value is larger than Number.MAX_SAFE_INTEGER")
         }
-        return long.toNumber();
+        if (num < ${bytes.globalThis}.Number.MIN_SAFE_INTEGER) {
+          throw new ${bytes.globalThis}.Error("Value is smaller than Number.MIN_SAFE_INTEGER")
+        }
+        return num;
```

Comment:
long.gt is actually not necessary to perform the check. I tested it with the script below, comparing the previous version against this version with values surrounding MAX_SAFE_INTEGER, and select values up to the max value for signed 64 integers.

<details><summary>test.ts</summary>

```ts
import Long from "long";

// The current implementation in ts-proto, using long.gt() to check bounds
function currentLongToNumber(input: string): number {
  const long = Long.fromString(input);
  if (long.gt(globalThis.Number.MAX_SAFE_INTEGER)) {
    throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
  }
  return long.toNumber();
}

// A more simple implementation that uses numbers only to check bounds
function betterLongToNumber(input: string): number {
  const num = Number(input);
  if (num > Number.MAX_SAFE_INTEGER) {
    throw new globalThis.Error("Value is larger than Number.MAX_SAFE_INTEGER");
  }
  return num;
}

const MAX_SIGNED_INT64 = 9_223_372_036_854_775_807n;
const MAX_SAFE_INTEGER = BigInt(Number.MAX_SAFE_INTEGER);
const tests: bigint[] = [];

// Add tests around Number.MAX_SAFE_INTEGER
for (let i = -100; i < 100; i++) {
   tests.push(BigInt(i) + MAX_SAFE_INTEGER);
}

// Add tests up to MAX_SIGNED_INT64
for (let i = MAX_SAFE_INTEGER + 100n; i < MAX_SIGNED_INT64;) {
  tests.push(i);
  i = i + (MAX_SIGNED_INT64 - MAX_SAFE_INTEGER) / 1_000_000n;
}
tests.push(MAX_SIGNED_INT64);


// Make sure both functions behave the same for each test
for (const test of tests) {
  let a: unknown;
  try {
    a = currentLongToNumber(test.toString());
  } catch (e) {
    a = e;
  }
  let b: unknown;
  try {
    b = betterLongToNumber(test.toString());
  } catch (e) {
    b = e;
  }
  const ok = (a instanceof Error && b instanceof Error)
    ? a.message === b.message
    : a === b;
  if (!ok) {
    throw new Error(`case ${test}: ${String(a)} !== ${String(b)}`);
  }
}

console.log(tests.length, "tests passed");
```
</details>